test/plugin/test_buffer_file_chunk.rb

```diff
@@ -768,4 +771,64 @@ def gen_chunk_path(prefix, unique_id)
       end
     end
   end
+
+    test '#open passes decompressed data to block when compress is gzip' do
```

Comment:
What `#open` passes (to block) is not data, but io object to provide decompressed data.